### PR TITLE
[LAPACK] Add builder

### PR DESCRIPTION
**Pull request by ViralBShah**
Addresses #2657 

The main missing thing here is to make it depend on our own BLAS, ideally through LBT. I'm not quite sure how to make its cmake recipe pick a BLAS - help welcome.

### CODE REVIEW

File: L/LAPACK/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/lapack*
+
+mkdir build && cd build
+cmake .. \
+   -DJulia_PREFIX="$prefix" \
+   -DSingular_PREFIX="$prefix" \
```
Comment (giordano) [inline L/LAPACK/build_tarballs.jl:19]: Are these necessary?

File: L/LAPACK/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+cd $WORKSPACE/srcdir/lapack*
+
+mkdir build && cd build
+cmake .. \
+   -DCMAKE_INSTALL_PREFIX="$prefix" \
+   -DCMAKE_FIND_ROOT_PATH="$prefix" \
+   -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}" \
+   -DCMAKE_BUILD_TYPE=Release \
+   -DBUILD_SHARED_LIBS=ON \
+   -DBLAS_LIBRARIES="-L$prefix/lib -lblastrampoline.${dlext}"
```
Comment (giordano) [inline L/LAPACK/build_tarballs.jl:23]:
```suggestion
   -DBLAS_LIBRARIES="-L${libdir} \
   -lblastrampoline.${dlext}"
```

**Comment by ViralBShah** [comment]
@giordano I also want to build [ReLAPACK](https://github.com/HPAC/ReLAPACK). Should I just add it to this repo, or would it be better to have it separately built?

**Comment by giordano** [comment]
Yes, usually we prefer to keep things separate

**Comment by ViralBShah** [comment]
@giordano Ok to merge?

**Comment by giordano** [comment]
Yes, looks good

**Comment by ViralBShah** [comment]
We need to be ILP64 and have `64_` suffixes to be compatible easy replacement for the lapack we ship with openblas in julia.

**Pull request merged**